velox/core/SimpleFunctionMetadata.h

```diff
@@ -228,9 +276,11 @@ class ISimpleFunctionMetadata {
   virtual std::string getName() const = 0;
   virtual bool isDeterministic() const = 0;
   virtual int32_t reuseStringsFromArg() const = 0;
-  virtual std::shared_ptr<exec::FunctionSignature> signature() const = 0;
+  virtual uint32_t priority() const {
+    return UINT32_MAX; // Least priority.
```

Comment:
this comment is kind of confusing, this is the greatest possible value of priority (I know what you mean though).

also, if we're relying on this value anywhere it's almost certainly incorrect (it means we haven't computed the priority).

could this function just be pure virtual like the others in this class?

SimpleFunctionMetadata is the only implementation and it overrides this function

Reply:
I thought Koski extends this class?